packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -384,10 +384,10 @@ class _PagePosition extends ScrollPositionWithSingleContext implements PageMetri
   @override
   double? get page {
     assert(
-      !hasPixels || hasContentDimensions,
+      !hasPixels || hasContentDimensions || !haveDimensions,
       'Page value is only available after content dimensions are established.',
     );
-    return !hasPixels || !hasContentDimensions
+    return !hasPixels || !(hasContentDimensions || haveDimensions)
       ? null
       : _cachedPage ?? getPageFromPixels(clampDouble(pixels, minScrollExtent, maxScrollExtent), viewportDimension);
```

Comment:
I think we could tweak the boolean logic here to make it easier to follow:

```dart
return hasPixels && (hasContentDimensions || haveDimensions)
    ? _cachedPage ?? getPageFromPixels(clampDouble(...), viewportDimension)
    : null;
```

<br>

Alternatively, we could put a guard statement at the beginning to simplify both the `assert` and the `return` statement:

```dart
double? get page {
  if (!hasPixels) {
    return null;
  }
  assert(
    hasContentDimensions || !haveDimensions,
    'Page value is only available after content dimensions are established.',
  );
  return hasContentDimensions || haveDimensions
      ? _cachedPage ?? getPageFromPixels(clampDouble(...), viewportDimension)
      : null;
}
```